DSharpPlus/Net/Serialization/DiscordForumChannelJsonConverter.cs

```diff
@@ -22,56 +22,63 @@
 // SOFTWARE.
 
 using System;
+using System.Runtime.Serialization;
 using DSharpPlus.Entities;
 
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
-namespace DSharpPlus.Net.Serialization
-{
-    public class DiscordForumChannelJsonConverter : JsonConverter
-    {
-        public override bool CanWrite => false;
+namespace DSharpPlus.Net.Serialization;
 
-        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer) => throw new ();
+public class DiscordForumChannelJsonConverter : JsonConverter
+{
+    public override bool CanWrite => false;
 
-        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
-        {
-            var job = JObject.Load(reader);
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer) => throw new ();
 
-            DiscordChannel channel;
-            var channelType = job["type"].ToObject<ChannelType>();
+    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+    {
+        JObject job = JObject.Load(reader);
+        JToken typeToken;
+        bool hasType = job.TryGetValue("type", out typeToken);
 
-            if (channelType is ChannelType.GuildForum)
-            {
-                // Type erasure is almost unheard of in C#, but you never know...
-                var chn = new DiscordForumChannel();
-                serializer.Populate(job.CreateReader(), chn);
+        if (!hasType)
+        {
+            throw new SerializationException("Channel object lacks type - this should be reported to library developers");
```

Comment:
not convinced of the SerializationException here, particularly because it belongs to S.R.Serialization, which should... generally be avoided. how about Newtonsoft.Json.JsonException instead?

Reply:
Sounds good, i only searched for "Serialization"